src/services/upload/uploader.ts

```diff
@@ -53,6 +56,26 @@ export default async function uploader(
             return { fileUploadResult: FileUploadResults.ALREADY_UPLOADED };
         }
 
+        const sameFileInOtherCollection = findSameFileInOtherCollection(
+            existingFiles,
+            metadata,
+            collection.id
+        );
+
+        if (sameFileInOtherCollection) {
+            logUploadInfo(
+                `same file in other collection found for  ${fileNameSize}`
+            );
+            const resultFile = Object.assign({}, sameFileInOtherCollection);
+            resultFile.collectionID = collection.id;
+            await addToCollection(collection, [resultFile]);
```

Comment:
it would be better if we return the encrypted new file from addToCollection and return that forward, we won't need the new skipDecryption then